binderhub/builder.py

```diff
@@ -542,31 +590,33 @@ def _check_result(future):
                         BUILD_COUNT.labels(
                             status="failure", **self.repo_metric_labels
                         ).inc()
-
                 await self.emit(event)
 
-        # Launch after building an image
-        if not failed:
-            BUILD_TIME.labels(status="success").observe(
-                time.perf_counter() - build_starttime
-            )
-            BUILD_COUNT.labels(status="success", **self.repo_metric_labels).inc()
-            with LAUNCHES_INPROGRESS.track_inprogress():
-                await self.launch(provider)
-            self.event_log.emit(
-                "binderhub.jupyter.org/launch",
-                5,
-                {
-                    "provider": provider.name,
-                    "spec": spec,
-                    "ref": ref,
-                    "status": "success",
-                    "build_token": self._have_build_token,
-                    "origin": self.settings["normalized_origin"]
-                    if self.settings["normalized_origin"]
-                    else self.request.host,
-                },
-            )
+        if build_only_outcome:
+            return
+        else:
```

Comment:
A code style detail, feel free to mark as resolved without further ado if you think its nicer as it is!

I've grown to appreciate the pattern of returning and avoiding nesting inside an `else` section. In this case it would reduce the lines changed by indentation only, which provides a small upside for reviewers as well.

Current:

```python
if x:
    return
else:
    print("hi")
```

The considered option:

```python
if x:
    return

print("hi")
```